kubectl-openebs/cli/command/get/volume_list.go

```diff
@@ -45,64 +47,107 @@ func NewCmdGetVolume() *cobra.Command {
 		Long:    volumesListCommandHelpText,
 		Run: func(cmd *cobra.Command, args []string) {
 			openebsNs, _ := cmd.Flags().GetString("openebs-namespace")
-			util.CheckErr(RunVolumesList(cmd, openebsNs, args), util.Fatal)
+			casType, _ := cmd.Flags().GetString("cas-type")
+			casType = strings.ToLower(casType)
+			util.CheckErr(RunVolumesList(openebsNs, casType, args), util.Fatal)
 		},
 	}
 	return cmd
 }
 
 // RunVolumesList lists the volumes
-func RunVolumesList(cmd *cobra.Command, openebsNs string, vols []string) error {
-	k8sClient, err := client.NewK8sClient("")
-	util.CheckErr(err, util.Fatal)
-	if openebsNs == "" {
-		nsFromCli, err := k8sClient.GetOpenEBSNamespace(util.CstorCasType)
-		if err != nil {
-			//return errors.Wrap(err, "Error determining the openebs namespace, please specify using \"--openebs-namespace\" flag")
-			return errors.New("no cstor volumes found in the cluster")
+func RunVolumesList(openebsNs, casType string, vols []string) error {
+	k8sClient, err := client.NewK8sClient(openebsNs)
+	if err != nil {
+		return err
+	}
+	var nsMap map[string]string
+	// 0. Figure out the openebsNs & casType mess!!
+	if openebsNs == "" && casType != "" {
+		openebsNs, err := k8sClient.GetOpenEBSNamespace(casType)
+		if err == nil {
+			// TODO: Verbose log for this estimated namespace
+			k8sClient.Ns = openebsNs
+		} else {
+			_, _ = fmt.Fprintf(os.Stderr, "couldn't figure out openebs-namespace for cas-type=%s\n", casType)
```

Comment:
- Can we also add in the above message "please use --openebs-namespace flag to provide the ns"

Reply:
I really want to use klog here, but without the line number info, this would help me avoid the noise of the file-name:line number which is super helpful when debugging but not quite unless the user adds in a verbose flag(i.e. specifically asks for more qualified & quantified logs).